kyaml/runfn/runfn.go

```diff
@@ -507,7 +507,16 @@ func (r *RunFns) ffp(spec runtimeutil.FunctionSpec, api *yaml.RNode, currentUser
 	}
 
 	if r.EnableExec && spec.Exec.Path != "" {
-		ef := &exec.Filter{Path: spec.Exec.Path}
+		annotations := api.GetAnnotations()
+		wd, ok := annotations[kioutil.WorkingDirAnnotation]
```

Comment:
It occurs to me that this annotation could be set by the user, in theory. We _should_ be overwriting it in all cases, but should we validate it just in case? i.e. throw an error if it doesn't "look like" a value we set--it's relative, or root. Or would it be an option to use an unserialized field on the `FunctionSpec` to more definitively prevent it from being settable by the user?

Not caused by this PR, but I'm surprised to notice that we're not doing any validation of the exec path at all. There are several restrictions on the Starlark path above.

Reply:
I've added some validation to make sure its an absolute path, and not the root

Reply:
We shouldn't honor a user set value here.  The file path annotation should be set only by kustomize as it reads.  

That's what `AnnotateAll` does, so lgtm.

Regardless, sanity checks are always a good idea, because down the road someone will change the code.

Makes sense to fall back to cwd.

Reply:
Just noting that `AnnotateAll` will overwrite anything the user put in, so we're good.

Reply:
> Just noting that AnnotateAll will overwrite anything the user put in, so we're good.

Unless there's a code path that doesn't run that, which as Natasha pointed out, is true of `kustomize fn run`--if I'm understanding correctly, the annotation would currently be honored in that code path, which we don't want. 

> If so, what would that look like in the code? I may need some guidance as I've never seen that or tried to do that before.

I think it would necessarily mean not using an annotation for this, but instead setting the root as a field on one of the function-related structs that are used for configuration. I'm not confident in my understanding of these code paths, but maybe the current kustomization root could be an option passed to loader creation in `NewLoader` within `(b *Kustomizer) Run`, then `loadPlugin` could set it on a `FunctionSpec.WorkingDir` field?

Reply:
How important is it that `kustomize fn run` does not honor the annotation? What are the consequences of allowing the user to set the working dir for this case?

Reply:
@monopole @KnVerey the most recent commit takes a different approach, and passes down the working directory through parameters and fields so that we don't have to worry about the user ever being able to set it themselves. PTAL